examples/with-grafana/main.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/Trendyol/kafka-konsumer"
+	"strconv"
+	"sync/atomic"
+	"time"
+)
+
+type user struct {
+	ID   int
+	Name string
+}
+
+var messages = []user{
+	{ID: 1, Name: "foo"},
+	{ID: 2, Name: "bar"},
+	{ID: 3, Name: "baz"},
+	{ID: 4, Name: "qux"},
+	{ID: 5, Name: "fred"},
+}
+
+func main() {
+	// retryMap stores the number of retries for each message
+	var retryMap = make(map[int]int, len(messages))
+	for _, message := range messages {
+		retryMap[message.ID] = 0
+	}
+
+	// create new kafka producer
+	producer, _ := kafka.NewProducer(kafka.ProducerConfig{
+		Writer: kafka.WriterConfig{
+			Brokers: []string{"localhost:29092"},
+		},
+	})
+
+	// produce messages at 1 seconds interval
+	ticker := time.NewTicker(1 * time.Second)
```

Comment:
Shall we extract this section to another go file for simplicity?

Reply:
I think extracting this section makes the code more complex